test/factories/ansible_roles.rb

```diff
@@ -4,4 +4,16 @@
   factory :ansible_role do
     sequence(:name) { |n| "ansible_role_#{n}" }
   end
+
+  factory :host_ansible_role do
+    sequence(:position)
+    host
+    ansible_role
+  end
+
+  factory :hostgroup_ansible_role do
+    sequence(:position)
```

Comment:
If I'm correct, if you omit possition, acts_as_list will assign it automatically as the highest in given scope.
This would assign weirdly high possitions as it is rising with every usage of the factory.
```suggestion
    position { nil }
```